File: tests/fixtures/haskell/basics.hs
data User = User
    { userId :: Int
    , userName :: String
    , userEmail :: Maybe String
    }

greet :: User -> String
greet user = "Hello, " ++ userName user ++ "!"

main :: IO ()
main = do
    let user = User 1 "Alice" (Just "alice@example.com")
    putStrLn $ greet user

File: tests/fixtures/elixir/basics.ex
defmodule User do
  @moduledoc "User module"
  
  defstruct [:id, :name, :email]
  
  def new(name, email) do
    %__MODULE__{id: UUID.uuid4(), name: name, email: email}
  end
  
  def greet(%__MODULE__{name: name}) do
    "Hello, #{name}!"
  end
end